cocotb/share/makefiles/Makefile.pylib.Darwin

```diff
@@ -54,20 +49,7 @@ ifneq ($(NATIVE_ARCH),$(ARCH))
 	PYTHON_DYNLIBDIR:=$(subst 64,,$(PYTHON_DYNLIBDIR))
 endif
 
-# Since we don't know which modules we might need or whether the simulator
-# we're using passes the RTLD_GLOBAL to dlopen or whether the OS supports
-# is we simply link against all dynamic libraries in the Python installation
-ifneq (, $(shell which $(PYTHON_BIN)-config 2>/dev/null))
-	PY_CFG_EXE=$(PYTHON_BIN)-config
-else ifneq (, $(shell which $(PYTHON_BIN)$(PYTHON_MAJOR_VERSION)-config 2>/dev/null))
-	PY_CFG_EXE=$(PYTHON_BIN)$(PYTHON_MAJOR_VERSION)-config
-else ifneq (, $(shell which $(PYTHON_BIN)$(PYTHON_VERSION)-config 2>/dev/null))
-	PY_CFG_EXE=$(PYTHON_BIN)$(PYTHON_VERSION)-config
-else
-	$(error ERROR: Unable to find a valid python-config.)
-endif
-
-PYLIBS = $(shell $(PY_CFG_EXE) --libs)
+PYLIBS:=$(shell $(PYTHON_BIN) -c 'from __future__ import print_function; from distutils import sysconfig; import os; print("-l"+os.path.splitext(sysconfig.get_config_var("LIBRARY"))[0][3:])')
```

Comment:
According to the implementation of `python-config.py`, the implementation should be something like:
```
    elif opt in ('--libs', '--ldflags'):
        libs = []
        if '--embed' in opt_flags:
            libs.append('-lpython' + pyver + sys.abiflags)
        else:
            libpython = getvar('LIBPYTHON')
            if libpython:
                libs.append(libpython)
        libs.extend(getvar('LIBS').split() + getvar('SYSLIBS').split())

        # add the prefix/lib/pythonX.Y/config dir, but only if there is no
        # shared library in prefix/lib/.
        if opt == '--ldflags':
            if not getvar('Py_ENABLE_SHARED'):
                libs.insert(0, '-L' + getvar('LIBPL'))
        print(' '.join(libs))
```

Reply:
Maybe we should wait with all those changes until we can use `cocotb-config`.